Source/embeddings/Builder/EmbeddingClassBuilder.ts

```diff
@@ -68,16 +68,16 @@ export class EmbeddingClassBuilder<T> implements ICanBuildAndRegisterAnEmbedding
         }
         logger.debug(`Building embedding ${decoratedType.embeddingId} from type ${this._embeddingType.name}`);
 
-        const getCompareMethod = CompareDecoratedMethods.methodPerEmbedding.get(this._embeddingType);
+        const getCompareMethod = UpdateDecoratedMethods.methodPerEmbedding.get(this._embeddingType);
         if (getCompareMethod === undefined) {
-            logger.warn(`The embedding class ${this._embeddingType.name} must have a method decorated with @${compareDecorator.name} decorator`);
+            logger.warn(`The embedding class ${this._embeddingType.name} must have a method decorated with @${updateDecorator.name} decorator`);
             return;
         }
         const compareMethod = this.createCompareMethod(getCompareMethod);
 
         const getRemoveMethod = RemoveDecoratedMethods.methodPerEmbedding.get(this._embeddingType);
         if (getRemoveMethod === undefined) {
```

Comment:
```suggestion
        const getDeleteMethod = DeleteDecoratedMethods.methodPerEmbedding.get(this._embeddingType);
        if (getDeleteMethod === undefined) {
```

Let's also rename the `RemoveDecoratedMethods` to `DeleteDecoratedMethods` (or `DeletionDecoratedMethods`, whichever we chose earlier for the singular method)